packages/monaco/src/index.tsx

```diff
@@ -90,87 +86,122 @@ export function CodePanel(props: CodePanelProps) {
+            await typeCheck();
+
+            setTestEditorState(editor);
+          },
+          user: async (editor, monaco) => {
+            setMonacoInstance(monaco);
+
+            monaco.languages.typescript.typescriptDefaults.setCompilerOptions({
+              ...monaco.languages.typescript.typescriptDefaults.getCompilerOptions(),
+              strict: true,
+              target: monaco.languages.typescript.ScriptTarget.ESNext,
+              strictNullChecks: true,
+            });
+
+            monaco.languages.registerDocumentFormattingEditProvider(
+              'typescript',
+              PrettierFormatProvider,
+            );
+
+            setUserEditorState(editor);
+
+            const model = editor.getModel();
+
+            if (!model) {
+              throw new Error();
+            }
+
+            const ts = await (await monaco.languages.typescript.getTypeScriptWorker())(model.uri);
+
+            monaco.languages.registerInlayHintsProvider(
+              'typescript',
+              createTwoslashInlayProvider(monaco, ts),
+            );
+          },
+        }}
+        onChange={{
+          user: async (code) => {
+            if (!monacoInstance) return null;
+            setCode(code ?? '');
+            setLocalStorageCode(code ?? '');
+            const getTsWorker = await monacoInstance.languages.typescript.getTypeScriptWorker();
+
+            const mm = monacoInstance.editor.getModel(monacoInstance.Uri.parse(TESTS_PATH));
+            if (!mm) return null;
+
+            const tsWorker = await getTsWorker(mm.uri);
+
+            const errors = await Promise.all([
+              tsWorker.getSemanticDiagnostics(TESTS_PATH),
+              tsWorker.getSyntacticDiagnostics(TESTS_PATH),
+              Promise.resolve([]),
+              tsWorker.getCompilerOptionsDiagnostics(TESTS_PATH),
+            ] as const);
+
+            setTsErrors(errors);
+          },
+        }}
+      />
       <div
         className={clsx(
           {
             'justify-between': testEditorState,
           },
-          'sticky bottom-0 flex items-center justify-end p-2 dark:bg-[#1e1e1e]',
+          'sticky bottom-0 flex items-center justify-between p-2 dark:bg-[#1e1e1e]',
         )}
       >
-        {userEditorState && <VimStatusBar editor={userEditorState} />}
-        <div className="flex items-center justify-center gap-4">
+        <div className="flex items-center gap-4">
+          <Button
+            className="flex items-center gap-1"
+            variant="ghost"
+            size="sm"
+            onClick={() => {
+              setIsTestPanelExpanded((tp) => !tp);
+            }}
+          >
+            Tests
+            {isTestPanelExpanded ? <ChevronDown size={16} /> : <ChevronUp size={16} />}
+          </Button>
+          {hasFailingTest && <XCircle className="stroke-red-600 dark:stroke-red-300" />}
+          {!hasFailingTest && <CheckCircle2 className="stroke-green-600 dark:stroke-green-300" />}
```

Comment:
nitpick: ternary for `isTestPanelExpanded` but not for `hasFailingTest`?

Reply:
yep yep ill fix

Reply:
Solved!